include/SFML/Graphics/Rect.hpp

```diff
@@ -104,43 +104,19 @@ class Rect
     ////////////////////////////////////////////////////////////
     constexpr std::optional<Rect<T>> findIntersection(const Rect<T>& rectangle) const;
 
-    ////////////////////////////////////////////////////////////
-    /// \brief Get the position of the rectangle's top-left corner
-    ///
-    /// \return Position of rectangle
-    ///
-    /// \see getSize, getCenter
-    ///
-    ////////////////////////////////////////////////////////////
-    constexpr Vector2<T> getPosition() const;
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Get the size of the rectangle
-    ///
-    /// \return Size of rectangle
-    ///
-    /// \see getPosition, getCenter
-    ///
-    ////////////////////////////////////////////////////////////
-    constexpr Vector2<T> getSize() const;
-
     ////////////////////////////////////////////////////////////
     /// \brief Get the position of the center of the rectangle
     ///
     /// \return Center of rectangle
     ///
-    /// \see getSize, getPosition
-    ///
     ////////////////////////////////////////////////////////////
     constexpr Vector2<T> getCenter() const;
 
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    T left{};   //!< Left coordinate of the rectangle
-    T top{};    //!< Top coordinate of the rectangle
-    T width{};  //!< Width of the rectangle
-    T height{}; //!< Height of the rectangle
+    Vector2<T> position{}; //!< Position of the top-left corner of the rectangle
+    Vector2<T> size{};     //!< Size of the rectangle
```

Comment:
Have you considered adding `.getLeft()`, `.getTop()`, `.getWidth()`, `.getHeight()`, etc accessors to at least ease migration and provide some generally useful functions?

Reply:
Personally, I wouldn't go with this API. It's really not more difficult to write `position.x` / `position.y` / `size.x` / `size.y` than the respective getters, and using getters would create an asymmetrical API, unless you also add settings, which I would be even more more against.

Reply:
Counterpoint: having `.getRight()` and `.getBottom()` would be useful, but it would be weird to have those two and not left and top.